rllib/policy/policy.py

```diff
@@ -534,6 +537,152 @@ def _create_exploration(self) -> Exploration:
             framework=getattr(self, "framework", "tf"))
         return exploration
 
+    def _get_default_view_requirements(self):
+        """Returns a default ViewRequirements dict.
+
+        Note: This is the base/maximum requirement dict, from which later
+        some requirements will be subtracted again automatically to streamline
+        data collection, batch creation, and data transfer.
```

Comment:
Why not keep this as the empty dict and instead infer columns to add?

Reply:
It's not that easy:
1) Model may require some unknown inputs (such as prev-actions, etc..), which are only visible from the model's `self.inference_view_requirements`
2) Policy may need some standard inputs, such as "t", "episode_id" in its loss (or learn_on_batch) methods.

So we do need to fill in the standard values at first (then remove if not needed). The only exception are models extra-action-fetches, which probably shouldn't be in the initial dict and can be added after the model test-call.

Reply:
done.
So we are initially only adding the following base columns:
OBS
NEXT_OBS
ACTIONS
REWARDS
DONES
INFOS
EPS_ID
AGENT_INDEX
t (<- time step)

We then add the model's own inference requirements, including inferring some requirements from the model's init-state (done in the policy now, the model does not do this anymore).
Then we do the model forward test-pass and add the returned extra-action outs to the view reqs.
Then we call postprocessing and the loss, after which we erase all columns that are not needed (thereby differentiating between postprocessing and loss (some cols are only needed for postprocessing)).